### PR TITLE
Serve index.html for root requests

### PR DESCRIPTION
**Pull request by vendamere**
This implements the behavior described in issue #6

### CODE REVIEW

File: src/Servant/Static/TH/Internal/Api.hs [hunk 1]
```diff
@@ -21,8 +21,11 @@ fileTreeToApiType :: FileTree -> Q Type
 fileTreeToApiType (FileTreeFile filePath _) = do
   addDependentFile filePath
   MimeTypeInfo mimeT respT _ <- extensionToMimeTypeInfoEx filePath
-  let fileNameLitT = litT $ strTyLit $ takeFileName filePath
-  [t|$(fileNameLitT) :> Get '[$(mimeT)] $(respT)|]
+  let fileName = takeFileName filePath
+  let fileNameLitT = litT $ strTyLit fileName
+  case fileName of
+    "index.html" -> [t|Get '[$(mimeT)] $(respT) :<|> $(fileNameLitT) :> Get '[$(mimeT)] $(respT)|]
```
Comment (cdepillabout) [inline src/Servant/Static/TH/Internal/Api.hs:27]:
```suggestion
    -- We special case files called "index.html" and generate a type that serves on both
    -- the root, and under the path "index.hmlt".
    "index.html" -> [t|Get '[$(mimeT)] $(respT) :<|> $(fileNameLitT) :> Get '[$(mimeT)] $(respT)|]
```

File: src/Servant/Static/TH.hs [hunk 2]
```diff
@@ -22,10 +22,10 @@ Haskell web API that looks like this:
   dir\/
   ├── js
   │   └── test.js
-  └── hello.html
+  └── index.html
```
Comment (cdepillabout) [inline src/Servant/Static/TH.hs:25]:
```suggestion
  ├── hello.html
  └── index.html
```

File: src/Servant/Static/TH.hs [hunk 3]
```diff
@@ -22,10 +22,10 @@ Haskell web API that looks like this:
...
 @
 
-Here's the contents of @\"hello.html\"@ and @\"js\/test.js\"@:
+Here's the contents of @\"index.html\"@ and @\"js\/test.js\"@:
```
Comment (cdepillabout) [inline src/Servant/Static/TH.hs:28]:
```suggestion
Here's the contents of @\"hello.html\"@, @\"index.html\"@ and @\"js\/test.js\"@:
```

**Comment by cdepillabout** [comment]
Thanks for putting this together!

I released this to hackage: https://hackage.haskell.org/package/servant-static-th-1.0.0.0

This package has been quite stable for a while, so I decided to bump to 1.0.0.0.

**Comment by vendamere** [comment]
Thanks!

**Pull request merged**